apps/prairielearn/src/lib/editors.ts

```diff
@@ -1162,28 +1162,49 @@ export class QuestionModifyEditor extends Editor {
 }
 
 export class QuestionDeleteEditor extends Editor {
-  private question: Question;
+  private questions: Question[];
 
-  constructor(params: BaseEditorOptions) {
+  constructor(params: BaseEditorOptions & { questionBatch?: Question[] }) {
     super(params);
 
-    this.question = params.locals.question;
-    this.description = `Delete question ${this.question.qid}`;
+    if (params.questionBatch !== undefined) {
+      this.questions = params.questionBatch;
+    } else {
+      this.questions = [params.locals.question];
+    }
+
+    this.description =
+      this.questions.length === 1
+        ? `Delete question ${this.questions[0].qid}`
+        : `Delete questions ${this.questions.map((x) => x.qid).join(', ')}`;
   }
 
   async write() {
-    assert(this.question.qid, 'question.qid is required');
+    this.questions.forEach((question) => assert(question.qid, 'question.qid is required'));
 
     debug('QuestionDeleteEditor: write()');
-    await fs.remove(path.join(this.course.path, 'questions', this.question.qid));
-    await this.removeEmptyPrecedingSubfolders(
-      path.join(this.course.path, 'questions'),
-      this.question.qid,
-    );
+
+    await this.questions.forEach(async (question) => {
+      if (question.qid) {
+        //never will be false, but this is to placate typechecks
+        await fs.remove(path.join(this.course.path, 'questions', question.qid));
+        await this.removeEmptyPrecedingSubfolders(
+          path.join(this.course.path, 'questions'),
+          question.qid,
+        );
+      }
+    });
 
     return {
-      pathsToAdd: [path.join(this.course.path, 'questions', this.question.qid)],
-      commitMessage: `delete question ${this.question.qid}`,
+      pathsToAdd: this.questions
+        .map((question) =>
+          question.qid ? path.join(this.course.path, 'questions', question.qid) : undefined,
+        )
+        .filter((x) => x !== undefined), //never will be false, but this is to placate typechecks
```

Comment:
Heh, this is the third time we're doing this check. Could you build an array of guaranteed non-null QIDs at the start of this function and use that throughout to avoid these repeated checks?

Reply:
Done in 253c7cd.